runtime/src/main/java/org/corfudb/protocols/logprotocol/SMRRecordGarbageInfo.java

```diff
@@ -0,0 +1,80 @@
+package org.corfudb.protocols.logprotocol;
+
+
+import io.netty.buffer.ByteBuf;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.ToString;
+import org.corfudb.runtime.CorfuRuntime;
+import org.corfudb.util.serializer.ICorfuSerializable;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * SMREntryGarbageInfo maintains garbage information of one SMREntry. This SMREntry could be stand-alone or a part of
+ * MultiSMREntry. The existence of SMREntryGarbageInfo itself indicates the associated SMREntry has been identified
+ * as garbage. SMREntryGarbageInfo could be used individually or composites more complicated SMRGarbageInfo.
+ */
+@ToString(callSuper = true)
+@NoArgsConstructor
+public class SMRRecordGarbageInfo {
+
+    /**
+     * Global address that detected the associated SMREntry as garbage.
+     */
+    @Getter
+    private long detectorAddress;
```

Comment:
Can you change this to `markerAddress` as we discussed before?

Reply:
done